cdap-kafka/src/main/java/co/cask/cdap/kafka/run/KafkaServerMain.java

```diff
@@ -137,28 +130,30 @@ public void destroy() {
   }
 
   private Properties generateKafkaConfig(int brokerId, String zkConnectStr, String hostname, int port,
-                                         int numPartitions, int replicationFactor, String logDir) {
-    Preconditions.checkState(port > 0, "Port number is invalid.");
-    Preconditions.checkState(numPartitions > 0, "Num partitions should be greater than zero.");
-
+                                         CConfiguration cConf) {
     Properties prop = new Properties();
+
     prop.setProperty("broker.id", Integer.toString(brokerId));
     if (hostname != null) {
       prop.setProperty("host.name", hostname);
     }
     prop.setProperty("port", Integer.toString(port));
-    prop.setProperty("socket.send.buffer.bytes", "1048576");
-    prop.setProperty("socket.receive.buffer.bytes", "1048576");
-    prop.setProperty("socket.request.max.bytes", "104857600");
-    prop.setProperty("log.dir", logDir);
-    prop.setProperty("num.partitions", Integer.toString(numPartitions));
-    prop.setProperty("log.retention.hours", "24");
-    prop.setProperty("log.flush.interval.messages", "10000");
-    prop.setProperty("log.flush.interval.ms", "1000");
-    prop.setProperty("log.segment.bytes", "536870912");
     prop.setProperty("zookeeper.connect", zkConnectStr);
-    prop.setProperty("zookeeper.connection.timeout.ms", "1000000");
-    prop.setProperty("default.replication.factor", Integer.toString(replicationFactor));
+
+    Map<String, String> propConfigs = cConf.getKafkaConfigs();
+    for (Map.Entry<String, String> pair : propConfigs.entrySet()) {
+      String key = pair.getKey();
+      if(key.equals(KafkaConstants.ConfigKeys.PORT_CONFIG) || key.equals(KafkaConstants.ConfigKeys.ZOOKEEPER_NAMESPACE_CONFIG)
```

Comment:
Ok I think I'm just gonna remove the continue statement, because setting three properties again isn't the end of the world, and logically will look cleaner.